language/move-prover/src/bytecode_translator.rs

```diff
@@ -264,30 +264,50 @@ impl<'env> ModuleTranslator<'env> {
             separate(vec![type_args_str.clone(), args_str.clone()], ", ")
         );
         self.writer.indent();
-        let mut ctor_expr = "$MapConstValue($DefaultValue())".to_owned();
-        for field_env in struct_env.get_fields() {
-            let field_param =
-                &format!("{}", field_env.get_name().display(struct_env.symbol_pool()));
-            let type_check = boogie_well_formed_check(
-                self.module_env.env,
-                field_param,
-                &field_env.get_type(),
-                WellFormedMode::Default,
-            );
-            emit!(self.writer, &type_check);
-            ctor_expr = format!(
-                "{}[{} := {}]",
+        // $Vector is either represented using sequences or integer maps
+        if self.options.backend.vector_using_sequences {
+            //Using sequences as the internal representation
+            let mut ctor_expr = "$EmptyValueArray()".to_owned();
+            for field_env in struct_env.get_fields() {
+                let field_param =
+                    &format!("{}", field_env.get_name().display(struct_env.symbol_pool()));
+                let type_check = boogie_well_formed_check(
+                    self.module_env.env,
+                    field_param,
+                    &field_env.get_type(),
+                    WellFormedMode::Default,
+                );
+                emit!(self.writer, &type_check);
+                ctor_expr = format!("$ExtendValueArray({},{})", ctor_expr, field_param);
```

Comment:
Maybe this is a separate PR, but if extend is removed how would I handle this construction in this file? Should I collect all the elements and first and then build the sequence myself? Because if everything is inlined that's what the code currently does. Extend is concat applied to the original sequence and a unit constructor, which is how I'd end up constructing a sequence.